source/extensions/clusters/aggregate/cluster_lb.h

```diff
@@ -0,0 +1,147 @@
+#pragma once
+
+#include "envoy/upstream/cluster_manager.h"
+
+#include "common/upstream/load_balancer_impl.h"
+#include "common/upstream/upstream_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Aggregate {
+
+class AggregateLoadBalancerContext : public Upstream::LoadBalancerContext {
+public:
+  AggregateLoadBalancerContext(Upstream::LoadBalancerContext* context,
+                               Upstream::LoadBalancerBase::HostAvailability host_avail,
+                               uint32_t host_priority)
+      : context_(context), host_avail_(host_avail), host_priority_(host_priority) {
+    if (context_ == nullptr) {
+      context_ = new Upstream::LoadBalancerContextBase();
```

Comment:
If the `context` is not null, the `unique_ptr` will wrap the context from outside, and it will release the resource when the object is deconstructed. It will break if the caller (who creates the context) uses context.

Reply:
i would use two different fields here instead of doing manual memory management, using a `owned_context_` that is a unique ptr that gets allocated + returned if the provided context is null. WDYT? I imagine the extra overhead of the field is negligible

Reply:
I am ok with the solution. Added an extra field.